gdal/swig/python/osgeo/utils/gdal_calc.py

```diff
@@ -457,23 +665,32 @@ def main(argv):
 
     parser.add_option("--outfile", dest="outF", help="output file to generate or fill", metavar="filename")
     parser.add_option("--NoDataValue", dest="NoDataValue", type=float, help="output nodata value (default datatype specific value)", metavar="value")
-    parser.add_option("--type", dest="type", help="output datatype, must be one of %s" % list(DefaultNDVLookup.keys()), metavar="datatype")
+    parser.add_option("--hideNoData", dest="hideNoData", action="store_true", help="ignores the NoDataValues of the input rasters", metavar="value")
+    parser.add_option("--type", dest="type", help="output datatype, must be one of %s" % GDALDataTypeNames, metavar="datatype")
     parser.add_option("--format", dest="format", help="GDAL format for output file", metavar="gdal_format")
     parser.add_option(
         "--creation-option", "--co", dest="creation_options", default=[], action="append",
         help="Passes a creation option to the output format driver. Multiple "
         "options may be listed. See format specific documentation for legal "
         "creation options for each format.", metavar="option")
-    parser.add_option("--allBands", dest="allBands", default="", help="process all bands of given raster (A-Z)", metavar="[A-Z]")
+    parser.add_option("--allBands", dest="allBands", default="", help="process all bands of given raster (a-z, A-Z)", metavar="[a-z, A-Z]")
     parser.add_option("--overwrite", dest="overwrite", action="store_true", help="overwrite output file if it already exists")
     parser.add_option("--debug", dest="debug", action="store_true", help="print debugging information")
     parser.add_option("--quiet", dest="quiet", action="store_true", help="suppress progress messages")
     parser.add_option("--optfile", dest="optfile", metavar="optfile", help="Read the named file and substitute the contents into the command line options list.")
 
+    # parser.add_option("--color_table", dest="color_table", help="color table file name")
+    parser.add_option("--extent", dest="extent", type=str,
```

Comment:
Proposing ignore and fail as value of ``choices`` would be cleaner. See https://github.com/OSGeo/PROJ-data/blob/master/grid_tools/ntv2_to_gtiff.py#L72 . This should be available even in the deprecated optparse we use according to https://docs.python.org/2/library/optparse.html#optparse.Option.choices

Reply:
nice.
done.